sound/soc/sof/ipc4-pcm.c

```diff
@@ -457,18 +457,70 @@ static void ipc4_ssp_dai_config_pcm_params_match(struct snd_sof_dev *sdev, const
 	}
 }
 
+/*
+ * Fixup DAI link parameters for sampling rate based on
+ * DAI copier configuration.
+ */
+static int sof_ipc4_pcm_dai_link_fixup_rate(struct snd_sof_dev *sdev,
+					    struct snd_pcm_hw_params *params,
+					    struct sof_ipc4_copier *ipc4_copier)
+{
+	struct snd_interval *rate = hw_param_interval(params, SNDRV_PCM_HW_PARAM_RATE);
+	struct sof_ipc4_available_audio_format *available_fmt =
+		&ipc4_copier->available_fmt;
+	struct sof_ipc4_audio_format *audio_fmt =
+		&available_fmt->input_pin_fmts[0].audio_fmt;
+	unsigned int be_rate  = audio_fmt->sampling_frequency;
```

Comment:
this is one of those rare cases where I would _not_ initialize at declaration to make it easier to see what we are trying to do.
Actually, this is not that ugly (85 char long line):
```c
u32 be_rate = available_fmt->input_pin_fmts[0].audio_fmt.sampling_frequency;
struct sof_ipc4_audio_format *audio_fmt;
```
sampling_frequency is `uint32_t` after all.

Reply:
@ujfalusi I'll keep "unsigned int" as "params_rate(params)" returns unsigned int (sound/pcm.h). Mixing u32 and unsigned int seems a bit sloppy for little gain.